src/clr-angular/forms/range/range.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  Directive,
+  Optional,
+  ViewContainerRef,
+  Renderer2,
+  ElementRef,
+  Injector,
+  Self,
+  Input,
+  Host,
+} from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+import { WrappedFormControl } from '../common/wrapped-control';
+import { ClrRangeContainer } from './range-container';
+import { RangeControllerService } from './providers/range-controller.service';
+
+@Directive({ selector: '[clrRange]', host: { '[class.clr-range]': 'true' } })
+export class ClrRange extends WrappedFormControl<ClrRangeContainer> {
+  private _minValue: number = 0;
+  private _maxValue: number = 100;
+  private _step: number = 1;
+
+  @Input('min')
+  set min(val: number) {
+    if (val != this._minValue && val !== undefined) {
```

Comment:
Can we use the `this.rangeController.rangeMinValue` if we just want to track the change and remove the `_minValue` 

Maybe only `_step` is not part of `RangeControllerService` but this is also another question why is not part of th e `RangeControllerService`

Reply:
I'll use the `this.rangeController.rangeMinValue` to track the value. 
`_step` is not a part of the RangeControllerService because it is not used in width calculations that are made in the container. If you suggest, I can easily add it to the service.